server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/DelegateSetContract.scala

```diff
@@ -0,0 +1,515 @@
+         |					}
+         |				}
+         |			}, "0"
+         |		]
+         |	]
+         |}""".stripMargin
+
+    val response =  `given`
+      .header(ACCEPT.toString, ACCEPT_RFC8621_VERSION_HEADER)
+      .body(request)
+    .when
+      .post
+    .`then`
+      .statusCode(SC_OK)
+      .contentType(JSON)
+      .extract
+      .body
+      .asString
+
+    assertThatJson(response)
+      .isEqualTo(
+        s"""{
+           |	"sessionState": "2c9f1b12-b35a-43e6-9af2-0106fb53a943",
+           |	"methodResponses": [
+           |		[
+           |			"Delegate/set",
+           |			{
+           |				"accountId": "29883977c13473ae7cb7678ef767cbfbaffc8a44a6e463d971d23a65c1dc4af6",
+           |				"newState": "2c9f1b12-b35a-43e6-9af2-0106fb53a943",
+           |				"notCreated": {
+           |					"4f29": {
+           |						"type": "invalidArguments",
+           |						"description": "User nonexistuser@domain.tld does not exist"
+           |					}
+           |				}
+           |			},
+           |			"0"
+           |		]
+           |	]
+           |}""".stripMargin)
+  }
+
+  @Test
+  def delegateSetShouldBeIdempotent(server: GuiceJamesServer): Unit = {
+    val request =
+      s"""{
+         |	"using": ["urn:ietf:params:jmap:core", "urn:apache:james:params:jmap:delegation"],
+         |	"methodCalls": [
+         |		[
+         |			"Delegate/set", {
+         |				"accountId": "29883977c13473ae7cb7678ef767cbfbaffc8a44a6e463d971d23a65c1dc4af6",
+         |				"create": {
+         |					"4f29": {
+         |						"username": "andre@domain.tld"
+         |					},
+         |					"4f30": {
+         |						"username": "andre@domain.tld"
+         |					}
+         |				}
+         |			}, "0"
+         |		]
+         |	]
+         |}""".stripMargin
+
+    val response =  `given`
+      .header(ACCEPT.toString, ACCEPT_RFC8621_VERSION_HEADER)
+      .body(request)
+    .when
+      .post
+    .`then`
+      .statusCode(SC_OK)
+      .contentType(JSON)
+      .extract
+      .body
+      .asString
+    val delegationId = DelegationId.from(BOB, ANDRE).serialize
+
+    assertThatJson(response)
+      .isEqualTo(
+        s"""{
+           |	"sessionState": "2c9f1b12-b35a-43e6-9af2-0106fb53a943",
+           |	"methodResponses": [
+           |		["Delegate/set", {
+           |			"accountId": "29883977c13473ae7cb7678ef767cbfbaffc8a44a6e463d971d23a65c1dc4af6",
+           |			"newState": "2c9f1b12-b35a-43e6-9af2-0106fb53a943",
+           |			"created": {
+           |				"4f29": {
+           |					"id": "$delegationId"
+           |				},
+           |				"4f30": {
+           |					"id": "$delegationId"
+           |				}
+           |			}
+           |		}, "0"]
+           |	]
+           |}""".stripMargin)
+
+    awaitAtMostTenSeconds.untilAsserted(() =>
+      assertThat(server.getProbe(classOf[DelegationProbe]).getAuthorizedUsers(BOB).asJavaCollection)
+        .containsExactly(ANDRE))
+  }
+}
```

Comment:
We should have 2 tests:
- mix cases: one success + one fail
- support backReference

Reply:
backReference is not really meaningful to test now while we don't have Delegate/get or Delegate/set destroy yet (to combine). Otherwise, we could do some dummy back reference, but I prefer to defer it to later.

Reply:
Agree. This has limited testing value as this is independant from this change set.